core/src/SYCL/Kokkos_SYCL_DeepCopy.hpp

```diff
@@ -131,6 +131,157 @@ struct DeepCopy<Kokkos::Experimental::SYCLDeviceUSMSpace, Kokkos::HostSpace,
   }
 };
 
+template <>
+struct DeepCopy<Experimental::SYCLSharedUSMSpace,
+                Experimental::SYCLSharedUSMSpace, Kokkos::Experimental::SYCL> {
+  DeepCopy(void* dst, const void* src, size_t n) {
+    (void)
+        DeepCopy<Experimental::SYCLDeviceUSMSpace,
+                 Experimental::SYCLDeviceUSMSpace, Kokkos::Experimental::SYCL>(
+            dst, src, n);
+  }
+  DeepCopy(const Kokkos::Experimental::SYCL& instance, void* dst,
+           const void* src, size_t n) {
+    (void)
+        DeepCopy<Experimental::SYCLDeviceUSMSpace,
+                 Experimental::SYCLDeviceUSMSpace, Kokkos::Experimental::SYCL>(
+            instance, dst, src, n);
+  }
+};
```

Comment:
Did you consider doing
```C++
  template <>
  struct DeepCopy<Experimental::SYCLSharedUSMSpace,
                  Experimental::SYCLSharedUSMSpace, Kokkos::Experimental::SYCL>
      : DeepCopy<Experimental::SYCLDeviceUSMSpace,
                 Experimental::SYCLDeviceUSMSpace, Kokkos::Experimental::SYCL> {
  }

Reply:
Ok. Delegating makes it a few lines shorter.